core/src/main/java/org/elasticsearch/cluster/routing/AllocationId.java

```diff
@@ -148,12 +154,45 @@ public String toString() {
 
     @Override
     public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
-        builder.startObject("allocation_id");
-        builder.field("id", id);
+        builder.startObject();
+        builder.field(ID_KEY, id);
         if (relocationId != null) {
-            builder.field("relocation_id", relocationId);
+            builder.field(RELOCATION_ID_KEY, relocationId);
         }
         builder.endObject();
         return builder;
     }
+
+    public static AllocationId fromXContent(XContentParser parser) throws IOException {
+        XContentParser.Token token = parser.currentToken();
+        if (token == null) { // fresh parser? move to the first real token under object
```

Comment:
if this is done only in tests then the test should advance instead this makes 0 sense to me